robo-components/DeploymentTrait.php

```diff
@@ -793,4 +795,58 @@ public function deployNotify(string $pantheon_environment = 'qa', string $issue_
     }
   }
 
+  /**
+   * Uninstalls modules that are not specified in core.extension.yml.
+   *
+   * @param string $pantheon_environment
+   *   The Pantheon environment to run the commands against.
+   *
+   * @throws \Exception
+   */
+  public function uninstallExtraModules(string $pantheon_environment): void {
+    $pantheon_info = $this->getPantheonNameAndEnv();
+    $pantheon_terminus_environment = $pantheon_info['name'] . '.' . $pantheon_environment;
+
+    // Get the list of currently enabled modules.
+    $installed_modules_result = $this->taskExec("terminus remote:drush $pantheon_terminus_environment pm:list -- --status=enabled --format=json --type=module")
+      ->printOutput(FALSE)
+      ->run();
+
+    if ($installed_modules_result->getExitCode() !== 0) {
+      throw new \Exception("Failed to get the list of installed modules.");
+    }
+
+    $installed_modules_data = json_decode($installed_modules_result->getMessage(), TRUE);
+    $installed_modules = array_keys($installed_modules_data);
+
+    $core_extension_file = 'config/sync/core.extension.yml';
+    if (!file_exists($core_extension_file)) {
+      throw new \Exception("core.extension.yml file not found.");
+    }
+
+    $core_extensions = Yaml::parseFile($core_extension_file);
+    $required_modules = array_keys($core_extensions['module']);
+
+    // Determine extra modules and uninstall them.
+    $modules_to_uninstall = array_diff($installed_modules, $required_modules);
+
+    if (!empty($modules_to_uninstall)) {
+      try {
+        $uninstall_success = $this->taskExec("terminus remote:drush $pantheon_terminus_environment -- pm:uninstall " . implode(' ', $modules_to_uninstall) . " --yes")
+          ->run()
+          ->getExitCode();
+        $needs_revert = !$uninstall_success;
```

Comment:
```suggestion
        $needs_revert = $this->taskExec("terminus remote:drush $pantheon_terminus_environment -- pm:uninstall " . implode(' ', $modules_to_uninstall) . " --yes")
          ->run()
          ->getExitCode();
```

Exit code 0 is success.